### PR TITLE
Defining a common user-group to match all spoke-clusters

### PR DESCRIPTION
**Pull request by yue9944882**
so that we can grant extra common RBAC permissions to the CSR-generated certificates mounted by the spoke-agent

@qiujian16

### CODE REVIEW

File: pkg/spoke/hubclientcert/controller.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ const (
 	TLSCertFile = "tls.crt"
 
 	subjectPrefix         = "system:open-cluster-management:"
+	groupAllSpokeClusters = subjectPrefix + "spoke-clusters"
```
Comment (qiujian16) [inline pkg/spoke/hubclientcert/controller.go]: maybe spokeClustersGroup with a comment that all spoke is in this group?

**Review by qiujian16 — COMMENTED** [review]
/ok-to-test

**Comment by qiujian16** [comment]
/approve

@skeeey would you also take a look? It seems secure to me

File: pkg/hub/user/identity.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package user
+
+const (
+	// SubjectPrefix is a prefix for marking open-cluster-management users
+	SubjectPrefix = "system:open-cluster-management:"
+	// SpokeClustersGroup is a common group for all spoke clusters
+	SpokeClustersGroup = SubjectPrefix + "spoke-clusters"
```
Comment (elgnay) [inline pkg/hub/user/identity.go]: minor: Use "managedclusters" or "managed-clusters" instead of "spoke-clusters"

File: pkg/hub/csr/controller.go [hunk 3]
```diff
@@ -158,16 +159,20 @@ func isSpokeClusterClientCertRenewal(csr *certificatesv1beta1.CertificateSigning
 		return false
 	}
 
-	if len(x509cr.Subject.Organization) != 1 {
+	requestingOrgs := sets.NewString(x509cr.Subject.Organization...)
+	if requestingOrgs.Has(user.SpokeClustersGroup) { // optional common group for backward-compatibility
```
Comment (skeeey) [inline pkg/hub/csr/controller.go]: would you add some ut for this?

Comment (yue9944882) [reply pkg/hub/csr/controller.go]: test cases added covering the latest and the previous format of CSR

File: pkg/hub/user/identity.go (hunk 2 above)
Comment (yue9944882) [reply pkg/hub/user/identity.go]: done, dont have a strong preference between the names tho

**Pull request merged**